docs/introduction-to-smart-contracts.rst

```diff
@@ -563,9 +563,23 @@ idea, but it is potentially dangerous, as if someone sends Ether to removed
 contracts, the Ether is forever lost.
 
 .. warning::
-    From version 0.8.18 and up, the use of ``selfdestruct`` in both Solidity and Yul will trigger a
-    deprecation warning, since the ``SELFDESTRUCT`` opcode will eventually undergo breaking changes in behavior
-    as stated in `EIP-6049 <https://eips.ethereum.org/EIPS/eip-6049>`_.
+    From ``EVM >= Cancun`` onwards, ``selfdestruct`` will **only** send all Ether in the account to the given recipient and not destroy the contract.
+    However, when ``selfdestruct`` is called in the same transaction that creates the contract calling it,
+    the behaviour of ``selfdestruct`` before Cancun hardfork (i.e., ``EVM <= Shanghai``) is preserved and will destroy the current contract,
+    deleting any data, including storage keys, code and the account itself.
+    See `EIP-6780 <https://eips.ethereum.org/EIPS/eip-6780>`_ for more details.
+
+    The new behaviour is the result of a network-wide change that affects all contracts present on
+    the Ethereum mainnet and testnets.
+    It is important to note that this change is dependent on the EVM version of the chain on which
+    the contract is deployed.
+    The ``--evm-version`` setting used when compiling the contract has no bearing on it.
+
+    Also, note that the ``selfdestruct`` opcode has been deprecated in Solidity version 0.8.18,
+    as recommended by `EIP-6049 <https://eips.ethereum.org/EIPS/eip-6049>`_.
+    The deprecation is still in effect and the compiler will still emit warnings on its use.
+    Any use in newly deployed contracts is strongly discouraged even if the new behavior is taken into account.
+    Future changes to the EVM might further reduce the functionality of the opcode.
```

Comment:
@mehtavishwa30 Taking a final look at the rendered version, this whole section about `selfdestruct` actually strikes me as too advanced for "Introduction to Smart Contracts". Mentioning `selfdestruct` made sense when it was in common use, but now that it has all these warnings, it's no longer recommended for newbies to touch it. Maybe we should simplify it for new users?

We could delete everything from it and leave only a short paragraph saying that contracts cannot be deleted, only deactivated. We still have a copy of most of these warnings in "Units and Global Variables" so we could link there for anyone who wants the details. But in the intro we'd only be mentioning the fact that something called `selfdestruct` exists, but you should no be tempted to use it for deleting contracts since it's deprecated or something like that.

I mean, this is not necessary in this PR, I'm merging it as is, but could be done in a follow-up PR if you want to brush up the intro page a little.